lib/fauna/page.rb

```diff
@@ -0,0 +1,98 @@
+module Fauna
+  class Page
+    # Always an Array. Element type may vary.
+    attr_reader :data
+    # Nilable cursor for the previous page.
+    attr_reader :before
+    # Nilable cursor for the next page.
+    attr_reader :after
+
+    # Convert a raw response hash to a Page.
+    def self.from_hash(hash)
+      Page.new hash[:data], hash[:before], hash[:after]
+    end
+
+    def initialize(data, before = nil, after = nil)
+      @data = data
+      @before = before
+      @after = after
+    end
+
+    # Return a new Page whose data has had block applied to each element.
+    def map_data(&block)
+      Page.new data.map(&block), before, after
+    end
+
+    def to_s
+      "Page(#{data}, #{before}, #{after})"
+    end
+
+    def ==(other)
+      return false unless other.is_a? Page
+      data == other.data && before == other.before && after == other.after
+    end
+
+    alias_method :eql?, :==
+
+    # Like set_iterator but enumerates over pages rather than their content.
+    def self.page_iterator(client, set_query, params = {})
+      PageIter.new(client, set_query, params).to_enum
+    end
+
+    ##
+    # Enumerable that keeps getting new values in a set through pagination.
+    # For example:
+    #
+    #   iter = Page.set_iterator client, set_query, map: (Query.expr do
+    #     lambda do |ref|
+    #       select [:data], get(ref)
+    #     end
+    #   end)
+    #   puts iter.to_a
+    #
+    # +client+:: A Client.
+    # +set_query+:: Set query to paginate, such as Query.match.
+    # +params+:: All optional.
+    #            +:map+:: Query.lambda for mapping set elements.
+    #            +:page_size+:: Number of instances to be fetched per page.
+    def self.set_iterator(client, set_query, params = {})
```

Comment:
I'm looking for something complementary to `page_iterator`. Maybe `element_iterator`?